dask_cuda/proxify_device_objects.py

```diff
@@ -2,50 +2,49 @@
 import pydoc
 from collections import defaultdict
 from functools import partial
-from typing import List, MutableMapping, TypeVar
+from typing import List, MutableMapping, Optional, Tuple, TypeVar
 
 import dask
 from dask.utils import Dispatch
 
 from .proxy_object import ProxyObject, asproxy
 
 dispatch = Dispatch(name="proxify_device_objects")
-ignore_types = None
+incompatible_types: Optional[Tuple[type]] = None
 
 T = TypeVar("T")
 
 
-def _register_ignore_types():
-    """Lazy register types that shouldn't be proxified
+def _register_incompatible_types():
+    """Lazy register types that ProxifyHostFile should unproxify on retrieval.
 
-    It reads the config key "jit-unspill-ignore" (DASK_JIT_UNSPILL_IGNORE),
-    which should be a comma seperated list of types to ignore. The default
-    value is:
-        DASK_JIT_UNSPILL_IGNORE="cupy.ndarray"
-
-    Notice, it is not possible to ignore types explicitly handled by this
-    module such as `cudf.DataFrame`, `cudf.Series`, and `cudf.Index`.
+    It reads the config key "jit-unspill-incompatible"
+    (DASK_JIT_UNSPILL_INCOMPATIBLE), which should be a comma seperated
+    list of types. The default value is:
+        DASK_JIT_UNSPILL_INCOMPATIBLE="cupy.ndarray"
     """
-    global ignore_types
-    if ignore_types is not None:
+    global incompatible_types
+    if incompatible_types is not None:
         return  # Only register once
     else:
-        ignore_types = ()
+        incompatible_types = ()
 
-    ignores = dask.config.get("jit-unspill-ignore", "cupy.ndarray")
-    ignores = ignores.split(",")
+    incompatibles = dask.config.get("jit-unspill-incompatible", "cupy.ndarray")
+    incompatibles = incompatibles.split(",")
```

Comment:
Thanks for working on this @madsbk . 

I am curious, What about stuff like [cupyx.scipy.sparse](https://docs.cupy.dev/en/v9.0.0/reference/sparse.html#module-cupyx.scipy.sparse) and [CumlArray](https://github.com/rapidsai/cuml/blob/818bd2503a840f6f85190a8d3dd719a36504bdf8/python/cuml/common/array.py#L36-L46)  . 

Will this work with those too ?